projects/core/src/icon/icon.stories.ts

```diff
@@ -67,12 +68,100 @@ class AllIcons extends LitElement {
   }
 }
 
+@customElement('demo-icons-collection')
+class CollectionIcons extends LitElement {
+  @state() icons: string[] = [];
+
+  private _collection: string;
+  get collection(): string {
+    return this._collection;
+  }
+
+  @property({ type: String })
+  set collection(val: string) {
+    this._collection = val;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    import('@cds/core/icon').then(module => {
+      module.loadChartIconSet();
+      module.loadCommerceIconSet();
+      module.loadCoreIconSet();
+      module.loadEssentialIconSet();
+      module.loadMediaIconSet();
+      module.loadMiniIconSet();
+      module.loadSocialIconSet();
+      module.loadTechnologyIconSet();
+      module.loadTextEditIconSet();
+      module.loadTravelIconSet();
+
+      const collections: any = {
+        chart: module.chartCollectionIcons,
+        commerce: module.commerceCollectionIcons,
+        core: module.coreCollectionIcons,
+        essential: module.essentialCollectionIcons,
+        media: module.mediaCollectionIcons,
+        mini: module.miniCollectionIcons,
+        social: module.socialCollectionIcons,
+        technology: module.technologyCollectionIcons,
+        textEdit: module.textEditCollectionIcons,
+        travel: module.travelCollectionIcons,
+      };
+      this.icons = collections[this.collection].map((icon: IconShapeTuple) => ClarityIcons.getIconNameFromShape(icon));
```

Comment:
This block could be shortened/simplified. This also reduces the places to change if a new collection is added.

```typescript
const collections: { [key: string]: IconShapeTuple[] } = {
  chart: module.chartCollectionIcons,
  commerce: module.commerceCollectionIcons,
  core: module.coreCollectionIcons,
  essential: module.essentialCollectionIcons,
  media: module.mediaCollectionIcons,
  mini: module.miniCollectionIcons,
  social: module.socialCollectionIcons,
  technology: module.technologyCollectionIcons,
  textEdit: module.textEditCollectionIcons,
  travel: module.travelCollectionIcons,
};

const collection = collections[this.collection];
this.icons = collection.map(icon => ClarityIcons.getIconNameFromShape(icon));

ClarityIcons.addIcons(...collection);
```

Reply:
This is another shortened option:

```typescript
@property({ type: String }) collection:
  | 'chart'
  | 'commerce'
  | 'core'
  | 'essential'
  | 'media'
  | 'mini'
  | 'social'
  | 'technology'
  | 'textEdit'
  | 'travel';

connectedCallback() {
  super.connectedCallback();
  import('@cds/core/icon').then(module => {
    const collection = module[`${this.collection}CollectionIcons`];
    this.icons = collection.map(icon => ClarityIcons.getIconNameFromShape(icon));

    ClarityIcons.addIcons(...collection);
  });
}
```

Reply:
And with a bit of TypeScript magic, you could infer the `collection` property type.

```typescript
type CollectionNames<T> = T extends `${infer CollectionName}CollectionIcons` ? CollectionName : never;

@customElement('demo-icons-collection')
class CollectionIcons extends LitElement {
  @state() icons: string[] = [];

  @property({ type: String }) collection: CollectionNames<keyof typeof import('@cds/core/icon')>;

  ...
}
```